echidna/core/scale.py

```diff
@@ -0,0 +1,157 @@
+import echidna.core.spectra as spectra
+import numpy
+
+
+class Scale(object):
+    """ A class for scaling the parameter space of a spectra.
+
+    Attributes:
+      _scale_factor (float): The factor you want to scale a parameter by.
+    """
+
+    def __init__(self):
+        """ Initialise the Scale class.
+        """
+        self._scale_factor = 1.
+
+    def get_scale_factor(self):
+        """ Returns the scale factor.
+
+        Returns:
+          float: The scale factor.
+        """
+        return self._scale_factor
+
+    def set_scale_factor(self, scale_factor):
+        """ Sets the scale factor.
+
+        Args:
+          scale_factor (float): Value you wish to set the scale factor to.
+
+       Raises:
+         ValueError: If scale_factor is zero or below.
+        """
+        if scale_factor <= 0.:
+            raise ValueError("Scale factor must be positive and non-zero.")
+        self._scale_factor = scale_factor
+
+    def scale(self, spectrum, dimension, **kwargs):
+        """ Scales a given spectrum's dimension.
+
+        Args:
+          spectrum (float): The spectrum you want to scale.
+          dimension (string): The dimension of the spectrum you want to scale.
+          kwargs (dict): To passed to the interpolation function in
+            :class:`echidna.core.spectra.Spectra`
```

Comment:
Is it just e.g. `kind='cubic'` that you add here to pass to `Spectra.interpolate1d`?